tasks/helpers/responsehelper.js

```diff
@@ -0,0 +1,36 @@
+const responseHelper = `/**
+ * @exports
+ * @class Response
+*/
+export default class Response {
+    /**
+     * @static
+     * @param {request} req request object
+     * @param {response} res response object
+     * @param {message} message message
+     * @param {data}  data
+     * @param {status} status status code
+     * @returns {object} object
+    */
+    static successMessage(req, res, message, data, status) {
+      res.status(status).json({
+        message,
+        data,
+      });
+    }
+  
+    /**
+     * @static
+     * @param {request} req request object
+     * @param {response} res response object
+     * @param {message} message message
+     * @param {status} status status code
+     * @returns {object} object
+    */
+    static errorMessage(req, res, message, status) {
+      res.status(status).json({
+        message,
+      });
+    }
+  }`
```

Comment:
I think these two methods are similar, I suggest we have just one function for JSON structure like so:

``` 
const resJSON = (
   success: boolean, 
   status: number, 
   message: string, 
   data?: string[] | {[key: string]: any}
)  => ({ success, status, message, data })
```

And when we need to send a response we can use the function: 

```
// if error
res.status(status).json(resJSON(false, 500, 'An error has occurred')) 

// if success
res.status(status).json(resJSON(true, 200, 'Listings fetched successfully',  [{...whatever listings json is...}]))

```

Reply:
Oh yeah @its-tayo , I actually see they are similar,  I had not realized in the begining, what I have done is to remove class declaration and delete one function and made the `data`  to come as the last attribute such that we have something similar to this 

```
    const responseHelper = (req, res, message, status, data) => {
      res.status(status).json({
        message,
        data,
      });
    }

```

Reply:
Alright, I think we can also remove the req arg. I am also not sure if extracting responses into a helper function is the right approach, I feel it should only be the JSON in the response. What are your thoughts on this @hezronkimutai?

Reply:
Yeah sure, @its-tayo , we actually don't need that `req`.
 About extracting responses into a helper function. I haven't understood the statement well, maybe if you explain more it would be cool.

Reply:
@hezronkimutai sorry for the late response, I got caught up with work. So what you have currently is a function that returns a response (`res.status(...).json()`): 

```
const responseHelper = (req, res, message, status, data) => {
      res.status(status).json({
        message,
        data,
      });
    }
```

But what I am suggesting is to have a function that only returns the JSON in the response (only return content of `.json()`). Using your format above we'll have a function like this:

```
const responseHelper = (message, status, data) => {
      return {message, status, data};
}
```

The only difference between this and the first code block I posted 3 days ago is that I included a `success` field in my code block, so I had this:

```
const responseHelper = (
   success: boolean, 
   status: number, 
   message: string, 
   data?: string[] | {[key: string]: any}
)  => ({ success, status, message, data })
```

So in the controllers, we'll send responses like this:
```
// if error
res.status(status).json(responseHelper(false, 500, 'An error has occurred')) 

// if success
res.status(status).json(responseHelper(true, 200, 'Listings fetched successfully',  [{...whatever listings json is...}]))
```

Reply:
I totally agree with you @its-tayo.

Reply:
```javascript
const responseHelper = (
   success: boolean, 
   status: number, 
   message: string, 
   data?: string[] | {[key: string]: any}
)  => ({ success, status, message, data })

// can be added inside middleware for validation. I think so. But, in another fix probably.
```